readthedocs/doc_builder/base.py

```diff
@@ -86,37 +86,43 @@ def docs_dir(self, docs_dir=None, **__):
             docs_dir = checkout_path
         return docs_dir
 
-    def create_index(self, extension='md', **__):
-        """Create an index file if it needs it."""
+    def create_index(self, extension='md', force_index=False, **__):
+        """
+        Create an index file if it needs it.
+
+        If force_index is True and there isn't an index file,
+        one is created whether or not there is a README file.
+        """
         docs_dir = self.docs_dir()
 
         index_filename = os.path.join(
             docs_dir, 'index.{ext}'.format(ext=extension))
-        if not os.path.exists(index_filename):
-            readme_filename = os.path.join(
-                docs_dir, 'README.{ext}'.format(ext=extension))
-            if os.path.exists(readme_filename):
-                return 'README'
-            else:
-                index_file = open(index_filename, 'w+')
-                index_text = """
+        readme_filename = os.path.join(
+            docs_dir, 'README.{ext}'.format(ext=extension))
+
+        if os.path.exists(index_filename):
+            return 'index'
+        elif not force_index and os.path.exists(readme_filename):
+            return 'README'
+
+        index_file = open(index_filename, 'w+')
+        index_text = """
 
 Welcome to Read the Docs
 ------------------------
 
 This is an autogenerated index file.
 
-Please create an ``index.{ext}`` or ``README.{ext}`` file with your own content
-under the root (or ``/docs``) directory in your repository.
+Please create an ``index.{ext}`` or ``README.{ext}`` (Sphinx only) file with
```

Comment:
Why is this logic Sphinx only? Do we actually do it for Sphinx projects, but not mkdocs? Should we do it for Sphinx projects?

Reply:
I was thinking the same, I'm going to check if this works on Sphinx, if so, probably this `force_index` would be the default behavior.